### DISCUSSION TITLE
"Best Practices for Securing Secrets in Source Code Repositories?"

### DISCUSSION THREAD

**Discussion by iterm-cmd** [community/community#139973] [Code Security]
### Select Topic Area

Question

### Body

Hi everyone,

In situations where environment variables aren’t an option, what are the best practices for securely managing secrets in a public repository?

Looking forward to your thoughts!

**Comment by TiepBK9056** [comment]
To manage secrets in a public GitHub repository without environment variables:

Encryption: Store secrets in the repo but encrypt them. Use a secure service or key management system to decrypt at runtime.
Secret Management Services: Use external services like AWS Secrets Manager or HashiCorp Vault to store and retrieve secrets securely during runtime.
GitHub Actions Secrets: For GitHub workflows, store secrets in GitHub Actions Secrets, which are encrypted and secure.
These methods help avoid exposing secrets directly in the codebase.

**Reply by iterm-cmd** [reply]
Thanks a lot

**Comment by ebndev** [comment]
Please see [this comment](https://github.com/orgs/community/discussions/139972#discussioncomment-10798827) for reference to why this discussion was closed and locked. Thanks!
